src/cases/containers/tasks-container/tasks-container.component.ts

```diff
@@ -1,30 +1,50 @@
+import { HttpClient } from '@angular/common/http';
 import { Component, OnInit } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
 import { CaseView } from '@hmcts/ccd-case-ui-toolkit';
-import { first } from 'rxjs/operators';
-import { Caseworker } from '../../../work-allocation-2/models/dtos';
+import { of } from 'rxjs';
+import { first, mergeMap } from 'rxjs/operators';
+import { Caseworker, TaskList } from '../../../work-allocation-2/models/dtos';
 import { Task } from '../../../work-allocation-2/models/tasks';
-import { WorkAllocationCaseService } from '../../../work-allocation-2/services';
+import { CaseworkerDataService, WorkAllocationCaseService } from '../../../work-allocation-2/services';
 
 @Component({
   selector: 'exui-tasks-container',
   templateUrl: './tasks-container.component.html',
   styleUrls: ['./tasks-container.component.scss']
 })
 export class TasksContainerComponent implements OnInit {
+  public static CASE_TASKS_URL: string = '/workallocation2/case/task';
+
   public caseDetails: CaseView;
   public tasks: Task[] = [];
   public tasksRefreshed: boolean = false;
   public caseworkers: Caseworker[] = [];
   public warningIncluded: boolean;
 
-  constructor(private readonly waCaseService: WorkAllocationCaseService, private readonly route: ActivatedRoute) { }
+  constructor(private readonly http: HttpClient, private readonly waCaseService: WorkAllocationCaseService, private readonly route: ActivatedRoute, private readonly caseworkerService: CaseworkerDataService) { }
 
   public ngOnInit(): void {
-    this.tasks = this.route.snapshot.data.tasks.tasks as Task[];
+    // note: internal logic used to be stored in resolver - resolver removed for smoother navigation purposes
+    // i.e. navigating before loading
+    const caseId = this.route.snapshot.paramMap.get('cid');
+    const tasks$ = this.http.get<Task[]>(`${TasksContainerComponent.CASE_TASKS_URL}/${caseId}`);
```

Comment:
I think we need to create a layer of indirection here to simplify the code. I would suggest we create a `TaskService` instead of injecting the `HttpClient` in the container. The TaskService can have a method  called `getTasksByCaseId(caseId: string)`;

Reply:
We already have a tasks service - the reason why I didn't put this in a service was because I was unsure whether this should go in the current task service, the current case service or a new casetaskservice. The api call actually uses case/task/:caseId so that implies it should be case service. I will ask Uday

Reply:
Hi Kuda, as mentioned - will do this on separate PR as priority